### ISSUE TITLE
Multi-GPU Support

### ISSUE THREAD

**Issue by jamescolless** [dreamquark-ai/tabnet#223]
## Feature request

**What is the expected behavior?**

Training across multiple gpus with near linear-speedup

**What is motivation or use case for adding/changing the behavior?**

Tabnet is rather slow on large datasets, can take advantage of single node multi-gpu training paradigm to speed this up.

**How should this be implemented in your opinion?**

Pytorch supports distributed training via torch.nn.DataParallel

**Are you willing to work on this yourself?**

yes

**Comment by pmdy** [comment]
Yes, please. It is very slow even with a Tesla P10 GPU on Colab.

**Comment by Optimox** [comment]
I think it's doable, probably without much changes.

Feel free to open a PR!

**Comment by freddycct** [comment]
Still waiting...

**Comment by Optimox** [comment]
@freddycct if you keep waiting, you'll end up waiting. If you start doing I will be more than happy to help or answer your questions.

The amont of people having multiple GPUs on their single machine and working with tabular data seems small to me. I welcome those people to offer a solution.

**Comment by freddycct** [comment]
@Optimox Yep I totally agree with you.

**Comment by Optimox** [comment]
I still don't feel it's a top priority, if someone wants to implement this, reopen the issue and create a PR.
